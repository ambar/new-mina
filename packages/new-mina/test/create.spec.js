const path = require('path')
const globby = require('globby')
const execa = require('execa')

// async timeout
jest.setTimeout(100000)

const resolveRelative = path.resolve.bind(null, __dirname)
const execP = (cmd, options) =>
  execa.shell(cmd, Object.assign({stdio: 'inherit'}, options))

describe('create', () => {
  beforeEach(() => {
    process.chdir(__dirname)
  })

  it('should add new app', async () => {
    const appName = `my-app`
    const appDir = resolveRelative(appName)

    await execP(`rm -rf ${appName}`)
    const script = `file:${resolveRelative('../../mina-scripts')}`
    await execP(`node ../bin/new-mina --script ${script} ${appName}`)

    process.chdir(appDir)
    const getAppfiles = () => globby(`**/*.*`, {gitignore: true})
    const getDistfiles = () => globby(`dist/**/*.*`)
    expect(await getAppfiles()).toMatchSnapshot()

    await execP(`npm run build`)
    expect(await getDistfiles()).toMatchSnapshot()
  })
})
