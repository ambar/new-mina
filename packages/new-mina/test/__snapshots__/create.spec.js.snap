// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create should add new app 1`] = `
Array [
  "package.json",
  "postcss.config.js",
  "prettier.config.js",
  "yarn.lock",
  "src/app.mina",
  "src/project.config.json",
  "src/components/counter.mina",
  "src/pages/home.mina",
  "src/styles/utils.wxss",
]
`;

exports[`create should add new app 2`] = `
Array [
  "dist/app.js",
  "dist/app.json",
  "dist/app.wxss",
  "dist/common.js",
  "dist/project.config.json",
  "dist/components/counter.js",
  "dist/components/counter.json",
  "dist/components/counter.wxml",
  "dist/components/counter.wxss",
  "dist/pages/home.js",
  "dist/pages/home.json",
  "dist/pages/home.wxml",
  "dist/pages/home.wxss",
]
`;
